# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/src/renderer/mod.rs
# Commit: 0ef9b28ae108
# Full Hash: 0ef9b28ae108513855b5e855f469cd64e278f6e4
# Author: Nicolas Silva <nsilva@mozilla.com>
# Date: 2021-09-30 21:50:26
# Regressor Bug: 1733154
# File Overlap Count: 1
# Description:
#   Bug 1733154 - Dequeue all GL errors instead of just the last one. r=gfx-reviewers,gw
#   
#   According to https://www.khronos.org/opengl/wiki/OpenGL_Error glGetError pops from a queue of errors rather than only keeping track of the last error so we need to dequeue all errors when checking for them in the renderer.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D126956
# ==============================================================================

diff -r 3772cc85bffe -r 0ef9b28ae108 gfx/wr/webrender/src/renderer/mod.rs
--- a/gfx/wr/webrender/src/renderer/mod.rs	Thu Sep 30 10:11:18 2021 +0000
+++ b/gfx/wr/webrender/src/renderer/mod.rs	Thu Sep 30 10:28:34 2021 +0000
@@ -2370,12 +2370,18 @@
     }
 
     fn check_gl_errors(&mut self) {
-        let err = self.device.gl().get_error();
-        if err == gl::OUT_OF_MEMORY {
-            self.renderer_errors.push(RendererError::OutOfMemory);
+        loop {
+            let err = self.device.gl().get_error();
+            if err == gl::NO_ERROR {
+                break;
+            }
+
+            if err == gl::OUT_OF_MEMORY {
+                self.renderer_errors.push(RendererError::OutOfMemory);
+            }
+
+            // Probably should check for other errors?
         }
-
-        // Probably should check for other errors?
     }
 
     fn bind_textures(&mut self, textures: &BatchTextures) {
